src/Tools/dotnet-monitor/Egress/Configuration/EgressPropertiesProvider.cs

```diff
@@ -19,6 +21,20 @@ public EgressPropertiesProvider(IEgressPropertiesConfigurationProvider provider)
             _provider = provider;
         }
 
+        /// <inheritdoc/>
+        public IDictionary<string, string> GetAllProperties()
+        {
+            Dictionary<string, string> properties = new Dictionary<string, string>();
+            foreach (IConfigurationSection section in _provider.Configuration.GetChildren())
+            {
+                if (section.Exists())
```

Comment:
nit: Do you need to check that it exists if you just got it from `GetChildren` method?

Reply:
The docs say this can return a `IConfigurationSection` with a `null` parameter, so I just want to omit those if present.

Reply:
That's not in the docs for `IConfiguration.GetChildren`: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.iconfiguration.getchildren?view=dotnet-plat-ext-6.0

The `GetSection` method can return an empty subsection if the key doesn't exist (still not `null`), but you are not using that API here.

Reply:
Ahh, that is what I was reading

Reply:
Let me go remove that check then